pkg/iac/scanners/terraform/parser/resolvers/registry.go

```diff
@@ -136,6 +136,29 @@ func (r *registryResolver) Resolve(ctx context.Context, target fs.FS, opt Option
 	return filesystem, prefix, downloadPath, true, nil
 }
 
+func getPrivateRegistryTokenFromEnvVars(hostname string) (string, error) {
+	token := ""
+	asciiHostname, err := idna.ToASCII(hostname)
+	if err != nil {
+		return "", fmt.Errorf("Could not convert hostname %s to a punycode encoded ASCII string so cannot find token for this registry", hostname)
+	} else {
+		envVar := fmt.Sprintf("TF_TOKEN_%s", strings.ReplaceAll(asciiHostname, ".", "_"))
+		token = os.Getenv(envVar)
+
+		// Dashes in the hostname can optionally be converted to double underscores
+		if token == "" {
+			envVar = strings.ReplaceAll(envVar, "-", "__")
+			token = os.Getenv(envVar)
+		}
+
+		if token == "" {
+			return "", fmt.Errorf("No token was found for the registry at %s", hostname)
+		} else {
+			return token, nil
+		}
+	}
+}
+
```

Comment:
How about if we drop the trailing else?

```suggestion
func getPrivateRegistryTokenFromEnvVars(hostname string) (string, error) {
	token := ""
	asciiHostname, err := idna.ToASCII(hostname)
	if err != nil {
		return "", fmt.Errorf("Could not convert hostname %s to a punycode encoded ASCII string so cannot find token for this registry", hostname)
	}

	envVar := fmt.Sprintf("TF_TOKEN_%s", strings.ReplaceAll(asciiHostname, ".", "_"))
	token = os.Getenv(envVar)

	// Dashes in the hostname can optionally be converted to double underscores
	if token == "" {
		envVar = strings.ReplaceAll(envVar, "-", "__")
		token = os.Getenv(envVar)
	}

	if token == "" {
		return "", fmt.Errorf("No token was found for the registry at %s", hostname)
	}
	return token, nil
	
}

```

Reply:
Agree, added